File: py/produk.py
class Produk:
    def __init__(self, id_produk, nama, harga, stok):
        self.__id_produk = id_produk
        self.__nama = nama
        self.__harga = harga
        self.__stok = stok

    # Getter
    def get_id_produk(self): return self.__id_produk
    def get_nama(self): return self.__nama
    def get_harga(self): return self.__harga
    def get_stok(self): return self.__stok

    # Setter
    def set_nama(self, nama): self.__nama = nama
    def set_harga(self, harga): self.__harga = harga
    def set_stok(self, stok): self.__stok = stok

    # Method tampil data
    def tampilkan_info(self):
        print(f"ID: {self.__id_produk}")
        print(f"Nama: {self.__nama}")
        print(f"Harga: {self.__harga}")
        print(f"Stok: {self.__stok}")
        print("-" * 30)

File: java/Produk.java
public class Produk {
    private String idProduk;
    private String nama;
    private double harga;
    private int stok;

    // Constructor
    public Produk(String idProduk, String nama, double harga, int stok) {
        this.idProduk = idProduk;
        this.nama = nama;
        this.harga = harga;
        this.stok = stok;
    }

    // Getter
    public String getIdProduk() {
        return idProduk;
    }

    public String getNama() {
        return nama;
    }

    public double getHarga() {
        return harga;
    }

    public int getStok() {
        return stok;
    }

    // Setter
    public void setNama(String nama) {
        this.nama = nama;
    }

    public void setHarga(double harga) {
        this.harga = harga;
    }

    public void setStok(int stok) {
        this.stok = stok;
    }

    // Method tampil data
    public void tampilkanInfo() {
        System.out.println("ID: " + idProduk);
        System.out.println("Nama: " + nama);
        System.out.println("Harga: " + (long) harga); // biar ga pakai notasi ilmiah
        System.out.println("Stok: " + stok);
        System.out.println("-----------------------------");
    }
}
